publish.sh

```diff
@@ -1,62 +1,156 @@
 #!/bin/bash
 
-set -ex
+set -e
 
-KUBEVIRTCI_TAG=$(date +"%y%m%d%H%M")-$(git rev-parse --short HEAD)
-export KUBEVIRTCI_TAG
+export KUBEVIRTCI_TAG=$(date +"%y%m%d%H%M")-$(git rev-parse --short HEAD)
 
 TARGET_REPO="quay.io/kubevirtci"
 TARGET_KUBEVIRT_REPO="quay.io/kubevirt"
 TARGET_GIT_REMOTE="https://kubevirt-bot@github.com/kubevirt/kubevirtci.git"
 
-# Build gocli
-(cd cluster-provision/gocli && make container)
-docker tag ${TARGET_REPO}/gocli ${TARGET_REPO}/gocli:${KUBEVIRTCI_TAG}
+function detect_cri() {
+    if podman ps >/dev/null 2>&1; then echo podman; elif docker ps >/dev/null 2>&1; then echo docker; fi
+}
 
-# Provision all base images
-(cd cluster-provision/centos8 && ./build.sh)
-(cd cluster-provision/centos9 && ./build.sh)
+export CRI_BIN=${CRI_BIN:-$(detect_cri)}
+export DRY_RUN=${DRY_RUN:-false}
 
-# Provision all clusters
+PREV_KUBEVIRTCI_TAG=$(curl -sL https://storage.googleapis.com/kubevirt-prow/release/kubevirt/kubevirtci/latest?ignoreCache=1)
 CLUSTERS="$(find cluster-provision/k8s/* -maxdepth 0 -type d -printf '%f\n')"
-for i in ${CLUSTERS}; do
-    cluster-provision/gocli/build/cli provision cluster-provision/k8s/$i
-    docker tag ${TARGET_REPO}/k8s-$i ${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}
+IMAGES_TO_BUILD=()
+IMAGES_TO_RETAG=()
 
-    cluster-provision/gocli/build/cli provision cluster-provision/k8s/$i --slim
-    docker tag ${TARGET_REPO}/k8s-$i ${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-slim
-done
+function run_provision_manager() {
+  # Run provision manager to determine which images to build
+  TARGETS=$(./hack/pman/pman.sh)
+  echo "INFO: Provision manager suggests to build the following providers: ${TARGETS[@]}"
+
+  declare -A targets_map
+  for target in ${TARGETS[@]}; do
+    targets_map[$target]=$target
+  done
+
+  for i in ${CLUSTERS}; do
+    if [ "${targets_map[$i]}" != "" ]; then
+      IMAGES_TO_BUILD+=($i)
+      echo "INFO: $i will be provisioned"
+    else
+      IMAGES_TO_RETAG+=($i)
+      echo "INFO: $i will be retagged ($PREV_KUBEVIRTCI_TAG)"
+    fi
+  done
+
+  if [ $DRY_RUN == true ]; then
+    exit 0
+  fi
+}
 
-# Provision alpine container disk and tag it
-(cd cluster-provision/images/vm-image-builder && ./create-containerdisk.sh alpine-cloud-init)
-docker tag  alpine-cloud-init:devel ${TARGET_REPO}/alpine-with-test-tooling-container-disk:${KUBEVIRTCI_TAG}
-docker tag  alpine-cloud-init:devel ${TARGET_KUBEVIRT_REPO}/alpine-with-test-tooling-container-disk:devel
+function build_gocli() {
+  (cd cluster-provision/gocli && make container)
+  ${CRI_BIN} tag ${TARGET_REPO}/gocli ${TARGET_REPO}/gocli:${KUBEVIRTCI_TAG}
+}
 
-# Push all images
+function build_centos8_base_image() {
+  for target in ${IMAGES_TO_BUILD[@]}; do
+    if [[ ! "$target" =~ "centos9" ]]; then
```

Comment:
Note: we are intending to remove the `-centos9` suffix in the 1.27 provider version, so this and the one below will stop working then. Can we simplify and just build both base images?

Reply:
(see comment below)